From b3db9c9a2a1e7ed46a4cbf193b26a777e6d2b999 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 2 Dec 2014 21:03:35 -0500
Subject: [kernel] sched: Use new KABI macros

Message-id: <1417554215-238434-11-git-send-email-dzickus@redhat.com>
Patchwork-id: 101097
O-Subject: [RHEL7 PATCH 10/10] kabi, sched.h: Use new KABI macros
Bugzilla: 1164383
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1164383
Upstream: RHEL only

Convert some of the kern/sched/sched.h GENKSYMS usage to use the new kabi macros.

Reviewed-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Don Zickus <dzickus@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 910a7bc..7f24e53 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -8,6 +8,8 @@
 #include <linux/tick.h>
 #include <linux/slab.h>
 
+#include <linux/rh_kabi.h>
+
 #include "cpupri.h"
 #include "cpuacct.h"
 
@@ -382,10 +384,8 @@ struct root_domain {
   */
  cpumask_var_t rto_mask;
  struct cpupri cpupri;
-#ifndef __GENKSYMS__
  /* Indicate more than one runnable task for any CPU */
- bool overload;
-#endif
+ RH_KABI_EXTEND(bool overload)
 };
 
 extern struct root_domain def_root_domain;
-- 
1.7.1